packages/core-mobile/app/new/features/swap/hooks/useEvmSwap.ts

```diff
@@ -57,25 +64,32 @@ export const useEvmSwap = (): {
         }
       }
 
+      if (!isSwapUseMarkrBlocked && onUpdate === undefined) {
+        throw new Error('onUpdate is required when swap use markr is enabled')
+      }
```

Comment:
you can move this into the getQuote of the provider once you have the strategy pattern
```
const MarkrProvider: SwapProvider = {
  async getQuote(params) {
    const { onUpdate, ...rest } = params
    if (!onUpdate) {
      throw new Error('Markr provider requires an onUpdate callback')
    }
    ...
  },
}
```